### PR TITLE
Add third column to specific emplacements table describing relationship

### LINKED ISSUE #5502: Claim in documentation not supported by code
http://symfony.com/doc/current/cookbook/bundles/best_practices.html

screenshot from 2015-07-08 13 11 14

I cannot find any reference in Symfony2 code to EventListener being a specific emplacement (I understand that it may be a convention) whereas the Command, Controller and Resources subdirectories have some degree of auto-wiring functionality during compiler passes hence the specific emplacement claim.

Command: src/Symfony/Component/HttpKernel/Bundle/Bundle.php <-- wires up Command directory
Controller: src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php <-- relies on a specific emplacement of Controller to generate short names
etc...

My understanding of "specific emplacement" is that it is mandated due to the way Symfony2 processes bundles being loaded into the AppKernel. Am I missing something here? 

You could put listeners under any directory and it makes no difference. It's just poor style. This is not true for the other directories listed in the screenshot above.

### PR DESCRIPTION
**Pull request by rcousens**
The standard Symfony2 framework has mandatory requirements about specific emplacements, and then others that are considered best practice by convention. This commit makes the documentation clearer in order to provide a developer with reasonable expectations about the consequences of altering the default directory structure. Additionally, newer developers may be less intimidated and concerned by whether they are compromising any auto-wiring functionality by deviating from the information provided.

Fixes #5502

### CODE REVIEW

**Comment by javiereguiluz** [comment]
:+1: thank you @rcousens.

The only thing that I don't like is the column title (`Relationship`). I propose to change it by `Location Type` or something like that. But before doing any change, let's see what do doc managers think about this.

**Comment by rcousens** [comment]
Sure thing, I also think it could use a little extra padding as it looks a bit cramped! Will wait for feedback and make the final changes then.

File: cookbook/bundles/best_practices.rst [hunk 1]
```diff
@@ -117,20 +117,20 @@ files are going to be part of the repository.
...
+Commands                         ``Command/``                  Mandatory
+Controllers                      ``Controller/``               Mandatory
+Service Container Extensions     ``DependencyInjection/``      Mandatory
+Event Listeners                  ``EventListener/``            Convention
+Model classes [1]                ``Model/``                    Convention
+Configuration                    ``Resources/config/``         Mandatory
+Web Resources (CSS, JS, images)  ``Resources/public/``         Mandatory
+Translation files                ``Resources/translations/``   Mandatory
+Templates                        ``Resources/views/``          Mandatory
+Unit and Functional Tests        ``Tests/``                    Convention
```
Comment (WouterJ) [inline cookbook/bundles/best_practices.rst]: in some degree, this is mandatory too (as it's configured in the phpunit config file).

**Comment by rcousens** [comment]
Updated as per feedback

**Comment by javiereguiluz** [comment]
:+1: @rcousens thanks for your patience and for making all the asked changes! This is ready to be merged :)

PS: @WouterJ when merging this, please add two whitespaces as separation between the second and the new third column. Thanks!

**Comment by rcousens** [comment]
@WouterJ I fixed up the spacing :) All good! 

Thank you for letting me improve the documentation, I remember starting with Symfony2 almost three years ago now and the amount of internal anxiety I had over "specific emplacements" made me hesitant to structure bundles in a way that best expressed the domain. Knowing what is mandatory and what is optional is somewhat liberating when you finally work it out.

Cheers :+1:

**Comment by xabbuh** [comment]
Hm, thinking about this using only "Location" sounds weird to me. I would prefer "Location Type" or something like that instead.

File: cookbook/bundles/best_practices.rst [hunk 2]
```diff
@@ -117,20 +117,20 @@ files are going to be part of the repository.
...
+Commands                         ``Command/``                   Mandatory
+Controllers                      ``Controller/``                Mandatory
+Service Container Extensions     ``DependencyInjection/``       Mandatory
+Event Listeners                  ``EventListener/``             Convention
+Model classes [1]                ``Model/``                     Convention
+Configuration                    ``Resources/config/``          Mandatory
+Web Resources (CSS, JS, images)  ``Resources/public/``          Mandatory
+Translation files                ``Resources/translations/``    Mandatory
+Templates                        ``Resources/views/``           Mandatory
+Unit and Functional Tests        ``Tests/``                     Mandatory
```
Comment (stof) [inline cookbook/bundles/best_practices.rst]: this is not really mandatory. It is perfectly fine to put them elsewhere (even outside the prod code of your bundle) as long as you configure your testing tool accordingly (and it is perfectly fine to use another tool than PHPUnit, in which case tests would be elsewhere)

**Comment by rcousens** [comment]
I will get this fixed up again this weekend. Cheers. Any preference on the third column header name?

Location vs Location Type?

**Comment by WouterJ** [comment]
Personally, I find the values of the last column too long. I propose to change it to use `Yes`/`No` as value:

sf-best-practices-mandatory

Or leave the column empty, unless the option is mandatory:

sf-best-practices-mandatory

**Comment by javiereguiluz** [comment]
@rcousens I've created #6616 to fix your work applying the comments made by @Wouter. I've reused your original commits, so you'll get full credit for your work. Thanks!

**Comment by xabbuh** [comment]
I am closing here in favour of #6616. Thank you for starting this change @rcousens!

**Pull request closed without merging**